src/SoftDeletes/TrashedDirective.php

```diff
@@ -29,28 +35,62 @@ public static function definition(): string
      */
     public function handleBuilder($builder, $value): object
     {
-        if ($builder instanceof Relation) {
-            $model = $builder->getRelated();
-        } elseif ($builder instanceof ScoutBuilder) {
-            $model = $builder->model;
-        } elseif ($builder instanceof EloquentBuilder) {
-            $model = $builder->getModel();
-        } else {
-            throw new \Exception('Can not get model from builder of class: '.get_class($builder));
+        if (! $builder instanceof EloquentBuilder) {
+            throw new Exception('Can not get model from builder of class: '.get_class($builder));
         }
+        $model = $builder->getModel();
 
-        SoftDeletesServiceProvider::assertModelUsesSoftDeletes(
-            get_class($model),
-            self::MODEL_MUST_USE_SOFT_DELETES
-        );
+        $this->assertModelUsesSoftDeletes($model);
 
-        if (! isset($value)) {
+        if ($value === null) {
             return $builder;
         }
 
-        $trashedModificationMethod = "{$value}Trashed";
-        $builder->{$trashedModificationMethod}();
+        /** @var Builder&SoftDeletes $builder */
+        switch ($value) {
+            case 'with':
+                $builder->withTrashed();
+
+            case 'only':
+                $builder->onlyTrashed();
+
+            case 'without':
+                $builder->withoutTrashed();
+
+            default:
+                throw new InvalidArgument('Unexpected value for Trashed filter');
+        }
 
         return $builder;
     }
+
+    public function handleScoutBuilder(ScoutBuilder $builder, $value)
+    {
+        $model = $builder->model;
+
+        $this->assertModelUsesSoftDeletes($model);
+
+        if ($value === null) {
+            return $builder;
+        }
+
+        switch ($value) {
+            case 'with':
```

Comment:
Does without not work with scout?

Reply:
Apparently not, since the builder lacks the method. I see this discovery as a big win of this PR 💪

Reply:
Yeah, this is a perfect example of how awesome this change is!